frontend/src/scenes/actions/ActionEdit.tsx

```diff
@@ -61,27 +65,49 @@ export function ActionEdit({ action: loadedAction, id, onSave, temporaryToken }:
 
     return (
         <div className="action-edit-container">
-            <PageHeader title={id ? 'Editing action' : 'Creating action'} buttons={deleteAction} />
+            <PageHeader
+                title={
+                    <EditableField
+                        name="name"
+                        value={action.name || ''}
+                        placeholder="Name this action"
+                        onChange={(name) => {
+                            setAction({ ...action, name })
+                            setEdited(!!name)
+                        }}
+                        persistEditMode={!id}
+                        minLength={1}
+                        maxLength={400} // Sync with action model
+                        data-attr={`action-name-${id ? 'edit' : 'create'}`}
+                        saveButtonText="Set"
+                    />
+                }
+                caption={
+                    <EditableField
+                        multiline
+                        name="description"
+                        value={action.description || ''}
+                        placeholder="Description (optional)"
+                        onSave={(description) => {
+                            setAction({ ...action, description })
+                            setEdited(!!description)
+                        }}
+                        data-attr="action-description"
+                        compactButtons
+                        maxLength={600} // No limit on backend model, but enforce shortish description
+                        isGated={!hasAvailableFeature(AvailableFeature.INGESTION_TAXONOMY)}
+                        saveButtonText="Set"
+                    />
```

Comment:
It's a bit confusing that the name field here is persistently editable when creating the action, but the description field isn't. Would be more intuitive if they behaved the same.